src/storage/src/hummock/cache.rs

```diff
@@ -305,66 +414,68 @@ impl<K: PartialEq + Default, T> LruCacheShard<K, T> {
         value: T,
         last_reference_list: &mut Vec<T>,
     ) -> *mut LruHandle<K, T> {
-        let mut handle = if let Some(mut h) = self.object_pool.pop() {
-            h.key = key;
-            h.value = Some(value);
+        self.evict_from_lru(charge, last_reference_list);
+        if self.usage.load(Ordering::Relaxed) + charge > self.capacity && self.strict_capacity_limit
```

Comment:
I just thought whether `strict_capacity_limit` is necessary because if a data does not enter the cache, it will still hold by some executor and the next read  will cause another remote storage request because of cache miss.

Reply:
I think it's a trade off between using more memory and staying safe from OOM. For block cache I think it is necessary to have a `strict_capacity_limit` to ensure that we don't use over-sized memory for the cache. 

Maybe we can borrow the idea of high-low threshold? For `insert` we just make sure it won't go over high threshold and when we release a handle, we evict it from cache if the usage has gone over low threshold.

Reply:
I have remove strict capacity limit in https://github.com/singularity-data/risingwave/pull/1994 and this pr may cause a lot of conflict....